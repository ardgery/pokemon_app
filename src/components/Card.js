import React,{useContext,useEffect,useReducer} from 'react';
import { Link } from 'react-router-dom';
import { PokemonContext } from 'contexts/PokemonContext';
import { MyPokemonContext } from 'contexts/MyPokemonContext';
import mypokemonReducer from 'reducers/mypokemonReducer';

export default function Card({id,name,nickname,isList,cardIndex,totalOwned}) {
    // const {dispatchMyPokemon} = useContext(MyPokemonContext);
    const [mypokemon, dispatchMyPokemon] = useReducer( mypokemonReducer ,[],()=>{
        const localData = localStorage.getItem('mypokemon');
        return localData ? JSON.parse(localData) : [];
    });
    function ownedText(){
        if(parseInt(totalOwned)>0) return <p>Owned : {totalOwned}</p>;
        else return <p>You don't own this pokemon</p>;
    }
    function removePokemon(e){
        e.preventDefault();
        dispatchMyPokemon({
            type: "REMOVE_FROM_MY_LIST",
            nickname:nickname
        })
    }
    return (
        <div className={"card "+ ((cardIndex % 4== 0)?'mr-0':'')}>
            <Link to={`/pokemon/${id}`} className="imgWrapper"><img src={"https://pokeres.bastionbot.org/images/pokemon/"+id+'.png'} width="70%" alt=""/></Link>
            <div className="nameWrapper"><Link to={`/pokemon/${id}`} className="pokemon_name">{name}</Link></div>
            {isList && ownedText()}
            {nickname && (<p>Nickname : {nickname}</p>)}
            {nickname && (<button onClick={(e)=>removePokemon(e)}>Remove Pokemon</button>)}
        </div>
    )
}
