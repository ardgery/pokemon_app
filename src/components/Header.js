import React,{useState} from 'react';
import 'styles/pages/header.scss';
import { Link } from 'react-router-dom';

export default function Header() {
    const [activeMenu,setActiveMenu]=useState(1);
    return (
        <div className="header">
            <div className="menu">
                <Link to="/" className={activeMenu==1?'active':''} onClick={()=>setActiveMenu(1)}>Pokemon List</Link>
                <Link to="/mypokemon" className={activeMenu==2?'active':''} onClick={()=>setActiveMenu(2)}>My Pokemon</Link>
            </div>
        </div>
    )
}
