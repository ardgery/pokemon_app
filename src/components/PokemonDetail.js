import React,{useContext, useState} from 'react';
import 'styles/pages/pokemon_detail.scss';
import { GET_POKEMON_DETAIL } from 'graphqlquery/Queries';
import { useQuery } from '@apollo/client';
import { PokemonContext } from 'contexts/PokemonContext';
import Modal from 'react-modal';
import Loading from 'images/loading.gif';
import useWindowDimensions from 'components/util/useWindowDimensions';


export default function PokemonDetail() {
    const {width } = useWindowDimensions();
    function getPokemonIdFromUrl(){
        var url = new URL(window.location.href);
        let hrefSplit = url.href.split('/')
        let idPokemon = parseInt(hrefSplit[hrefSplit.length-1]);
        return idPokemon;
    }
    let id = getPokemonIdFromUrl();
    const { data } = useQuery(GET_POKEMON_DETAIL, {
        variables: { id }
    });
    const { dispatch,dispatchMyPokemon,mypokemon } = useContext(PokemonContext);
    const [nickname, setNickname] = useState('');
    const [modalVisible,setModalVisible] = useState(false);
    const customStyles = {
        content : {
          top       : '0',
          left      : '0',
          right     : '0',
          bottom    : '0',
          margin    : 'auto',
          width     : width < 482? 300 : 400,
          height    : 200,
          padding   :width < 482? 0 : 20,
        },
        overlay: {
            background : 'rgba(0,0,0,0.8)',
            zIndex:1
        }
    };
    const [initModal,setInitModal] = useState(true);
    const [loading,setLoading] = useState(false);
    const [result,setResult] = useState(false);
    const [isSuccess,setIsSuccess] = useState(false);
    const [resultMessage,setResultMessage] = useState('');
    const [inputMessage,setInputMessage] = useState('');
    const [finalResult,setFinalResult] = useState(false);

    function checkExistNickname(newNickname){
        var existNickname = false;
        for(let i =0;i<mypokemon.length;i++){
            if (mypokemon[i].nickname == newNickname) {
                existNickname = true;
                break;
            }
        }
        return existNickname;  
    }
    function getPokemon(){
        setInitModal(false);
        setLoading(true);
        setResult(false);
        setFinalResult(false);

        setTimeout(()=>{
            setLoading(false);
            setResult(true);
        },1250)
        
        let success = Math.random() < 0.5;

        if(success) {
            setIsSuccess(true)
            setResultMessage("Success Get this Pokemon! Please give a Nickname to your new Pokemon.")
        } else {
            setIsSuccess(false)
            setResultMessage("Failed to Catch this Pokemon! Note that probability to get a Pokemon is 50%.")
        }
    }
    function closeForm(){
        setResult(false);
        setInitModal(true);
        setResultMessage("");
        setModalVisible(false)
    }
    function validatePokemon(e){
        e.preventDefault();
        if(nickname==''){
            setInputMessage("Pokemon Nickname Cannot be Empty.") 
        }else if(checkExistNickname(nickname)){
            setInputMessage("Nickname is already exist on your owned pokemon") 
        }else{
            setLoading(true);
            setResult(false);
            setFinalResult(true);
            dispatch({
                type: "UPDATE_OWNED_POKEMON",
                payload: {id:id-1, nickname:nickname}
            });
            dispatchMyPokemon({
                type: "ADD_TO_MY_LIST",
                payload: {
                    id:id,
                    name:data.pokemonQuery.name,
                    nickname:nickname,
                }
            })
            setTimeout(()=>{
                setLoading(false);
                setResultMessage("Pokemon has been added to your List!")
                setFinalResult(true);
                setResult(true);
            },500)    
            
            setTimeout(()=>{
                setLoading(false);
                setResult(false);
                setFinalResult(false);
                closeForm();
            },1500)    
        }
    }

    return ( 
        <div className="detailWrapper">
            <div className="detailInside">

                {
                    data && (
                        <div>
                            <div className="imgWrapper">
                                <img src={"https://pokeres.bastionbot.org/images/pokemon/"+id+'.png'} width="300px" alt=""/>
                            </div>
                            <div className="detailColumns">
                                <h1>{data && data.pokemonQuery.name}</h1>
                                <div className="detailPokemon">
                                    <h2>Types</h2>
                                    <div>
                                        {
                                            data  && data.pokemonQuery.types.map((v,i)=>{
                                                return (
                                                    <p key={i}>{v.name}</p>
                                                )
                                            })
                                        }
                                    </div>
                                    <h2 className="headingMoves">Moves</h2>
                                    <div className="moves">
                                        {
                                            data  && data.pokemonQuery.moves.map((v,i)=>{
                                                if(i<3){
                                                    return (
                                                        <p key={i}>{v.name}</p>
                                                    )
                                                }
                                            })
                                        }
                                    </div>
                                </div>
                            </div> 
                        </div>
                    )
                }

                {!data && (<div className="loadingWrapper"><img src={Loading} alt=""/></div>)}


            </div>
            {data && (<button onClick={()=>setModalVisible(true)}>Catch this Pokemon!</button>)}
            <div>
                <Modal 
                    isOpen={modalVisible}
                    contentLabel="Minimal Modal Example"
                    ariaHideApp={false}
                    style={customStyles}
                >
                    <div className="modalInside">
                        { initModal && (
                                <div>
                                    <h2>Are you sure you want to catch this Pokemon?</h2>
                                    <div className="modalButtonWrapper">
                                        <button onClick={()=>getPokemon()}>Yes</button>
                                        <button className="closeForm" onClick={()=>closeForm()}>Cancel</button>
                                    </div>
                                </div>
                        )}
                        {result && (
                            <div>
                                <h2 className={isSuccess?'success':'danger'}>{resultMessage}</h2>
                                {isSuccess && !finalResult && (
                                    <form className="addNicknameForm">
                                        <p className="danger">{inputMessage}</p>
                                        <div className="inputWrapper">
                                            <input autoFocus placeholder="type nickname here..." type="text" onChange={(e)=>{setNickname(e.target.value)}} value={nickname}/>
                                            <button type="submit" onClick={(e)=>validatePokemon(e)}>Confirm</button>
                                        </div>
                                    </form>
                                )}
                                {!isSuccess && (
                                    <div className="modalButtonWrapper">
                                        <button onClick={()=>getPokemon()}>Try Again</button>
                                        <button className="closeForm" onClick={()=>closeForm()}>Close Form</button>
                                </div>
                                )}
                            </div>
                        )}
                        {loading && (<img src={Loading} alt="" width="120" />)}
                    </div>
                </Modal>
            </div>

        </div>
    )
}
