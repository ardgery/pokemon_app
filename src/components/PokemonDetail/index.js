import React,{useContext, useState,useReducer} from 'react';
import 'styles/pages/pokemon_detail.scss';
import { GET_POKEMON_DETAIL } from 'graphqlquery/Queries';
import { useQuery } from '@apollo/client';
import { MyPokemonContext } from 'contexts/MyPokemonContext';
import Modal from 'react-modal';
import Loading from 'images/loading.gif';
import useWindowDimensions from 'components/util/useWindowDimensions';
import mypokemonReducer from 'reducers/mypokemonReducer';

export function GetPokeChildren({txt}){
    return(
        <p>{txt}</p>
    )
}


export function GetPoke({drill}){
    return(
        <div>
            <p className="aca">{drill}</p>
            <GetPokeChildren txt="ehehe" className="hoba" />
        </div>
        
    )
    // if(par1){
    //     if(par2){
    //         return true;
    //     }
    // }
    // return false;
}

function setModalContentVisible(status){
    const [visible,setVisible] = useState(false);
    if(status) setVisible(true);
    return visible;
}

export function RenderModalContent(modalVisible,setModalVisible,data,id){
    const {width } = useWindowDimensions();
    const [initModal,setInitModal] = useState(true);
    const [loading,setLoading] = useState(false);
    const [result,setResult] = useState(false);
    const [isSuccess,setIsSuccess] = useState(false);
    const [resultMessage,setResultMessage] = useState('');
    const [inputMessage,setInputMessage] = useState('');
    const [finalResult,setFinalResult] = useState(false);
    // const { dispatchMyPokemon,mypokemon } = useContext(MyPokemonContext);
    const [mypokemon] = useReducer( mypokemonReducer ,[],()=>{
        const localData = localStorage.getItem('mypokemon');
        return localData ? JSON.parse(localData) : [];
    });
    const [nickname, setNickname] = useState('');

    const customStyles = {
        content : {
          top       : '0',
          left      : '0',
          right     : '0',
          bottom    : '0',
          margin    : 'auto',
          width     : width < 482? 300 : 400,
          height    : 200,
          padding   :width < 482? 0 : 20,
        },
        overlay: {
            background : 'rgba(0,0,0,0.8)',
            zIndex:1
        }
    };

    function closeForm(){
        setResult(false);
        setInitModal(true);
        setResultMessage("");
        setModalVisible(false)
    }
    function checkExistNickname(newNickname){
        var existNickname = false;
        for(let i =0;i<mypokemon.length;i++){
            if (mypokemon[i].nickname == newNickname) {
                existNickname = true;
                break;
            }
        }
        return existNickname;  
    }
    function getSuccess(){
        return Math.random() < 0.5;
    }
    function getPokemon(isSuccess){
        setInitModal(false);
        setLoading(true);
        setResult(false);
        setFinalResult(false);

        setTimeout(()=>{
            setLoading(false);
            setResult(true);
        },1250)

        if(isSuccess) {
            setIsSuccess(true)
            setResultMessage("Success Get this Pokemon! Please give a Nickname to your new Pokemon.")
            return true;
        } else {
            setIsSuccess(false)
            setResultMessage("Failed to Catch this Pokemon! Note that probability to get a Pokemon is 50%.")
            return false;
        }
    }
    function validatePokemon(e){
        e.preventDefault();
        if(nickname==''){
            setInputMessage("Pokemon Nickname Cannot be Empty.") 
        }else if(checkExistNickname(nickname)){
            setInputMessage("Nickname is already exist on your owned pokemon") 
        }else{
            setLoading(true);
            setResult(false);
            setFinalResult(true);
            dispatchMyPokemon({
                type: "ADD_TO_MY_LIST",
                payload: {
                    id:id,
                    name:data.pokemonQuery.name,
                    nickname:nickname,
                }
            })
            setTimeout(()=>{
                setLoading(false);
                setResultMessage("Pokemon has been added to your List!")
                setFinalResult(true);
                setResult(true);
            },500)    
            
            setTimeout(()=>{
                setLoading(false);
                setResult(false);
                setFinalResult(false);
                closeForm();
            },1500)    
        }
    }

    return(
        <div>
            <Modal 
                isOpen={modalVisible}
                contentLabel="Minimal Modal Example"
                ariaHideApp={false}
                style={customStyles}
            >
                <div className="modalInside">
                    { initModal && (
                            <div>
                                <h2>Are you sure you want to catch this Pokemon?</h2>
                                <div className="modalButtonWrapper">
                                    <button onClick={()=>getPokemon(getSuccess())}>Yes</button>
                                    <button className="closeForm" onClick={()=>closeForm()}>Cancel</button>
                                </div>
                            </div>
                    )}
                    {result && (
                        <div>
                            <h2 className={isSuccess?'success':'danger'}>{resultMessage}</h2>
                            {isSuccess && !finalResult && (
                                <form className="addNicknameForm">
                                    <p className="danger">{inputMessage}</p>
                                    <div className="inputWrapper">
                                        <input autoFocus placeholder="type nickname here..." type="text" onChange={(e)=>{setNickname(e.target.value)}} value={nickname}/>
                                        <button type="submit" onClick={(e)=>validatePokemon(e,nickname)}>Confirm</button>
                                    </div>
                                </form>
                            )}
                            {!isSuccess && (
                                <div className="modalButtonWrapper">
                                    <button onClick={()=>getPokemon(getSuccess())}>Try Again</button>
                                    <button className="closeForm" onClick={()=>closeForm()}>Close Form</button>
                            </div>
                            )}
                        </div>
                    )}
                    {loading && (<img src={Loading} alt="" width="120" />)}
                </div>
            </Modal>
        </div>
    )
}

export function PokemonDetail() {
    function getPokemonIdFromUrl(){
        var url = new URL(window.location.href);
        let hrefSplit = url.href.split('/')
        let idPokemon = parseInt(hrefSplit[hrefSplit.length-1]);
        return idPokemon;
    }
    let id = getPokemonIdFromUrl();
    
    const { data } = useQuery(GET_POKEMON_DETAIL, {
        variables: { id }
    });
    const [modalVisible,setModalVisible] = useState(false);

    return ( 
        <div className="detailWrapper">
            <div className="detailInside">
                {
                    data && (
                        <div>
                            <div className="imgWrapper">
                                <img src={"https://pokeres.bastionbot.org/images/pokemon/"+id+'.png'} width="300px" alt=""/>
                            </div>
                            <div className="detailColumns">
                                <h1>{data && data.pokemonQuery.name}</h1>
                                <div className="detailPokemon">
                                    <h2>Types</h2>
                                    <div>
                                        {
                                            data  && data.pokemonQuery.types.map((v,i)=>{
                                                return (
                                                    <p key={i}>{v.name}</p>
                                                )
                                            })
                                        }
                                    </div>
                                    <h2 className="headingMoves">Moves</h2>
                                    <div className="moves">
                                        {
                                            data  && data.pokemonQuery.moves.map((v,i)=>{
                                                if(i<3){
                                                    return (
                                                        <p key={i}>{v.name}</p>
                                                    )
                                                }
                                            })
                                        }
                                    </div>
                                </div>
                            </div> 
                        </div>
                    )
                }

                {!data && (<div className="loadingWrapper"><img src={Loading} alt=""/></div>)}


            </div>
            {data && (<button onClick={()=>setModalVisible(true)}>Catch this Pokemon!</button>)}
            {RenderModalContent(modalVisible,setModalVisible,data,id)}
        </div>
    )
}