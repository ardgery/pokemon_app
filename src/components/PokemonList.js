import React from 'react';
import { GET_ALL_POKEMONS } from 'graphqlquery/Queries.js';
import { useQuery } from '@apollo/client';

export default function PokemonList() {
    const { loading, error, data } = useQuery(GET_ALL_POKEMONS);

    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error :(</p>;
      console.log("Data = ",data)
  
    return (
        <h1>Hehheeyyy</h1>
     );
}
