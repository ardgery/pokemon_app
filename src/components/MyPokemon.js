import React,{useEffect,useContext} from 'react';
import Card from 'components/Card';
import 'styles/pages/pokemon_list.scss';
import { PokemonContext } from 'contexts/PokemonContext';

export default function MyPokemon() {
    const {mypokemon} = useContext(PokemonContext);

    useEffect(()=>{
        console.log("myPokemon = ",mypokemon)
    },[])


    return (
        <div className="listWrapper">
            {mypokemon.length<1 && (
                <h1>You currently don't have any owned pokemon.</h1>
            )}
            <div className="list">
                {
                    mypokemon.map((v,i)=>{
                        return <Card key={i} name={v.name} nickname={v.nickname} id={v.id}/>
                    })
                }
            </div>
        </div>
    )
}
