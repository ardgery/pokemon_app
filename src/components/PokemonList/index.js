import React, {useContext,useEffect,useReducer} from 'react';
import Card from 'components/Card';
import 'styles/pages/pokemon_list.scss';
import { PokemonContext } from 'contexts/PokemonContext';
import { MyPokemonContext } from 'contexts/MyPokemonContext';
import Loading from 'images/loading.gif';
import { GET_ALL_POKEMONS } from 'graphqlquery/Queries';
import { useQuery } from '@apollo/client';
import mypokemonReducer from 'reducers/mypokemonReducer';

export function CountOwned(name,mypokemon){
    if(mypokemon){
        let returnValue = 0;
        for(let i=0;i<mypokemon.length;i++){
            if( mypokemon[i].name === name ){
                returnValue+=1;
            }
        }
        return returnValue;
    }
    return false;
}
export function PokemonList() {
    // const {loading,error,pokemons} = useContext(PokemonContext);
    const { loading, error, data } = useQuery(GET_ALL_POKEMONS);
    const [mypokemon] = useReducer( mypokemonReducer ,[],()=>{
        const localData = localStorage.getItem('mypokemon');
        return localData ? JSON.parse(localData) : [];
    });
    // const {mypokemon} = useContext(MyPokemonContext);
    useEffect(()=>{
        if(data){
            // console.log("pokemons = ",data.listpokemonQuery)
        }
        
    })
    if(error) return <div className="listWrapper loadingWrapper">Error on Fetching Data<h2>Error on Fetching Data</h2></div>;
    if(loading) return <div className="listWrapper loadingWrapper" role="loading">Loading...<img src={Loading} alt=""/></div>;
    return (
        <div className="listWrapper">
            <div className="list" id="list_pokemon">
                {
                    data.listpokemonQuery.map((v,i)=>{
                        return <Card key={i} name={v.name} id={i+1} cardIndex={i+1} totalOwned={CountOwned(v.name,mypokemon)} isList={true}  />
                    })
                }
            </div>
        </div>
    );
}
