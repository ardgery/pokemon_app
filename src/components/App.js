import React, { useEffect,useContext } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
// import { ApolloClient, InMemoryCache, ApolloProvider,createHttpLink } from '@apollo/client';
import Header from 'components/Header';
import Footer from 'components/Footer';
import { PokemonList, CountOwned } from 'components/PokemonList';
import { PokemonDetail } from 'components/PokemonDetail';
import MyPokemon from 'components/MyPokemon';
import PokemonContextProvider from 'contexts/PokemonContext';
import MyPokemonContextProvider from 'contexts/MyPokemonContext';



function App(){
    useEffect(()=>{
        // console.log(CountOwned("lulba",[{id:1,name:"soka",nickname:"lula"},{id:2,name:"cyka",name:"sata"}]))
    })
    return(
        <Router>
            <PokemonContextProvider>
                <MyPokemonContextProvider>
                    <div className="App">
                        <Header />
                        <Route exact path="/" component={PokemonList} />
                        <Route exact path="/mypokemon" component={MyPokemon} />
                        <Route exact path="/pokemon/:id" component={PokemonDetail} />
                        <Footer />
                    </div>
                </MyPokemonContextProvider>
            </PokemonContextProvider>
        </Router>
    )
}
export default App;
