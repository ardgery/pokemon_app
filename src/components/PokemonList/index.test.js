import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import TestRenderer from 'react-test-renderer';
import "regenerator-runtime/runtime";
import { PokemonList, CountOwned } from 'components/PokemonList';
import { GET_ALL_POKEMONS } from 'graphqlquery/Queries';
import { render, cleanup, findByTestId, findByText, waitFor, screen, fireEvent } from '@testing-library/react';
import { Link } from 'react-router-dom';
import { StaticRouter } from 'react-router';
const {act} = TestRenderer;

const mocksMyPokemon = [
  {
    name: "bulbasaur",
  },
  {
    name: "bulbasaur",
  },
  {
    name: "ivysaur",
  },
]

it('data matching test', async () => {
  const mocks = [
    {
      request: {
        query: GET_ALL_POKEMONS,
      },
      result: {
        data: {
          listpokemonQuery: [
            {
              name: "bulbasaur",
            },
            {
              name: "ivysaur",
            },
            {
              name: "venusaur",
            },
            {
              name: "charmander",
            },
            {
              name: "charmeleon",
            },
            {
              name: "charizard",
            },
            {
              name: "squirtle",
            },
            {
              name: "wartortle",
            },
            {
              name: "blastoise",
            },
            {
              name: "caterpie",
            },
            {
              name: "metapod",
            },
            {
              name: "butterfree",
            },
            {
              name: "weedle",
            },
            {
              name: "kakuna",
            },
            {
              name: "beedrill",
            },
            {
              name: "pidgey",
            },
            {
              name: "pidgeotto",
            },
            {
              name: "pidgeot",
            },
            {
              name: "rattata",
            },
            {
              name: "raticate",
            }
          ]
        }
      }
    }
  ]
  const component = TestRenderer.create(
    <StaticRouter>
      <MockedProvider mocks={mocks} addTypename={false}>
        <PokemonList />
      </MockedProvider>
    </StaticRouter>
  );
  const componentRoot = component.root;
  
  await act(()=> new Promise(resolve => setTimeout(resolve, 0)))
  
  expect(component.toJSON()).toMatchSnapshot()
  // const p = component.root.findByType('p');
  // const testInstance = component.root;
  // expect(testInstance.findByProps({className: "poggie"}).children).toEqual(['name:','bulbasaur']);
  // expect(p.children.join('')).toContain('hahaha');
});

it('count owned function test', () => {
  const countOwnedFunc = CountOwned("bulbasaur", mocksMyPokemon);
  expect(countOwnedFunc).toBe(2);
});