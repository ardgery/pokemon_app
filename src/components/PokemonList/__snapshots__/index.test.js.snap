// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data matching test 1`] = `
<div
  className="listWrapper"
>
  <div
    className="list"
    id="list_pokemon"
  >
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/1"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/1.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/1"
          onClick={[Function]}
        >
          bulbasaur
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/2"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/2.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/2"
          onClick={[Function]}
        >
          ivysaur
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/3"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/3.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/3"
          onClick={[Function]}
        >
          venusaur
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card mr-0"
    >
      <a
        className="imgWrapper"
        href="/pokemon/4"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/4.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/4"
          onClick={[Function]}
        >
          charmander
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/5"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/5.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/5"
          onClick={[Function]}
        >
          charmeleon
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/6"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/6.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/6"
          onClick={[Function]}
        >
          charizard
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/7"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/7.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/7"
          onClick={[Function]}
        >
          squirtle
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card mr-0"
    >
      <a
        className="imgWrapper"
        href="/pokemon/8"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/8.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/8"
          onClick={[Function]}
        >
          wartortle
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/9"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/9.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/9"
          onClick={[Function]}
        >
          blastoise
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/10"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/10.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/10"
          onClick={[Function]}
        >
          caterpie
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/11"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/11.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/11"
          onClick={[Function]}
        >
          metapod
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card mr-0"
    >
      <a
        className="imgWrapper"
        href="/pokemon/12"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/12.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/12"
          onClick={[Function]}
        >
          butterfree
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/13"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/13.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/13"
          onClick={[Function]}
        >
          weedle
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/14"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/14.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/14"
          onClick={[Function]}
        >
          kakuna
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/15"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/15.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/15"
          onClick={[Function]}
        >
          beedrill
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card mr-0"
    >
      <a
        className="imgWrapper"
        href="/pokemon/16"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/16.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/16"
          onClick={[Function]}
        >
          pidgey
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/17"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/17.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/17"
          onClick={[Function]}
        >
          pidgeotto
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/18"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/18.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/18"
          onClick={[Function]}
        >
          pidgeot
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card "
    >
      <a
        className="imgWrapper"
        href="/pokemon/19"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/19.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/19"
          onClick={[Function]}
        >
          rattata
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
    <div
      className="card mr-0"
    >
      <a
        className="imgWrapper"
        href="/pokemon/20"
        onClick={[Function]}
      >
        <img
          alt=""
          src="https://pokeres.bastionbot.org/images/pokemon/20.png"
          width="70%"
        />
      </a>
      <div
        className="nameWrapper"
      >
        <a
          className="pokemon_name"
          href="/pokemon/20"
          onClick={[Function]}
        >
          raticate
        </a>
      </div>
      <p>
        You don't own this pokemon
      </p>
    </div>
  </div>
</div>
`;
